testing/scenario_app/lib/src/platform_view.dart

```diff
@@ -878,131 +1034,159 @@ class PlatformViewScrollingUnderWidget extends Scenario
-        _valueInt32,
-        ..._to32(0), // LTR
-      ],
-      if (Platform.isAndroid && usesAndroidHybridComposition) ...<int>[
-        _valueString,
-        'hybrid'.length,
-        ...utf8.encode('hybrid'),
-        _valueTrue,
-        _valueString,
-        'direction'.length,
-        ...utf8.encode('direction'),
-        _valueInt32,
-        ..._to32(0), // LTR
-      ],
+      'hybrid'.length,
+      ...utf8.encode('hybrid'),
+      _valueTrue,
       _valueString,
-      'params'.length,
-      ...utf8.encode('params'),
-      _valueUint8List,
-      text.length,
-      ...utf8.encode(text),
-    ]);
-
-    dispatcher.sendPlatformMessage(
-      'flutter/platform_views',
-      message.buffer.asByteData(),
-      (ByteData? response) {
-        if (response != null &&
-            Platform.isAndroid &&
-            !usesAndroidHybridComposition) {
-          // Envelope.
-          _textureId = response.getUint8(0);
-        }
-      },
-    );
-  }
-
-  void _addPlatformViewToScene(
-    SceneBuilder sceneBuilder,
-    int viewId,
-    double width,
-    double height,
-  ) {
-    if (Platform.isIOS) {
-      sceneBuilder.addPlatformView(viewId, width: width, height: height);
-    } else if (Platform.isAndroid) {
-      if (usesAndroidHybridComposition) {
-        sceneBuilder.addPlatformView(viewId, width: width, height: height);
-      } else if (_textureId != null) {
-        sceneBuilder.addTexture(_textureId!, width: width, height: height);
+      'direction'.length,
+      ...utf8.encode('direction'),
+      _valueInt32,
+      ..._to32(0), // LTR
+    ],
+    _valueString,
+    'params'.length,
+    ...utf8.encode('params'),
+    _valueUint8List,
+    text.length,
+    ...utf8.encode(text),
+  ]);
+
+  dispatcher.sendPlatformMessage(
+    'flutter/platform_views',
+    message.buffer.asByteData(),
+    (ByteData? response) {
+      late int textureId;
+      if (response != null &&
+          Platform.isAndroid &&
+          !usesAndroidHybridComposition) {
+        // This is the texture ID.
+        assert(response.getUint8(0) == 0, 'expected envelope');
+        final int type = response.getUint8(1);
+        assert(type == 4, 'expected int64');
+        textureId = response.getInt64(2, Endian.host);
+      } else {
+        // There no texture ID.
+        textureId = -1;
       }
+      _createdPlatformViews[platformViewKey] = textureId;
+      created.complete();
+    },
+  );
+  await created.future;
+  addPlatformViewToSceneBuilder(
+    id,
+    sceneBuilder: sceneBuilder,
+    textureId: _createdPlatformViews[platformViewKey]!,
+    width: width,
+    height: height,
+  );
+}
+
+/// Adds the platform view to the scene builder.
+Future<void> addPlatformViewToSceneBuilder(
+  int id, {
+  required SceneBuilder sceneBuilder,
+  required int textureId,
+  double width = 500,
+  double height = 500,
+}) async {
+  final bool usesAndroidHybridComposition = scenarioParams['use_android_view'] as bool;
```

Comment:
scenarioParams['use_android_view'] is null on iOS and triggers a crash when trying to cast.

Reply:
done!